inference-engine/src/transformations/src/transformations/smart_reshape/mimic_set_batch_size.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <transformations/smart_reshape/mimic_set_batch_size.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::MimicSetBatchSize, "MimicSetBatchSize", 0);
+
+ngraph::pass::MimicSetBatchSize::MimicSetBatchSize() {
+    auto reshape_label = ngraph::pattern::wrap_type<opset5::Reshape>({pattern::any_input(), ngraph::pattern::wrap_type<opset5::Constant>()},
```

Comment:
We could simplify it in this way:

```
    auto reshape_label = pattern::wrap_type<opset5::Reshape>({pattern::any_input(pattern::has_static_dim(0)),
                                                              pattern::wrap_type<opse5::Constant>()},
         [](const Output<Node> &output) {
             const auto &output_rank = output.get_partial_shape().rank();
             return output_rank.is_static() && output_rank.get_length() > 1;
         });
```

I think we need to add more common predicates for shapes and ranks to simplify pattern development. I'll think about it :)